org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/CompletionPreferencePage.scala

```diff
@@ -0,0 +1,99 @@
+import org.scalaide.core.completion.PrefsBasedProposalRelevanceCfg.PShunnedPackages
+import org.scalaide.core.completion.PrefsBasedProposalRelevanceCfg.PUnpopularPackages
+
+object CompletionPreferencePage {
+  private class PackageGroupListEditor(name: String, groupTitle: String, dialogTitle: String, parent: Composite)
+      extends ListEditor(name, groupTitle, parent) {
+
+    getUpButton.setVisible(false)
+    getDownButton.setVisible(false)
+
+    protected def createList(arr: Array[String]): String = {
+      StringListSerializer.serialize(arr)
+    }
+
+    protected def parseString(str: String): Array[String] = {
+      StringListSerializer.deserialize(str).toArray
+    }
+
+    def getNewInputObject(): String = {
+      val dialog = new InputDialog(getShell, dialogTitle, "Enter a regular expression", null, PackageRxValidator)
+
+      if (dialog.open() != Window.OK) {
+        null
+      } else {
+        dialog.getValue
+      }
+    }
+  }
+
+  private class FavoritePackagesListEditor(parent: Composite)
+      extends PackageGroupListEditor(PFavoritePackages, "Favorite Packages", "Favorite package", parent)
+
+  private class PreferedPackagesListEditor(parent: Composite)
+      extends PackageGroupListEditor(PPreferredPackages, "Preferred Packages", "Preferred package", parent)
+
+  private class UnpopularPackagesListEditor(parent: Composite)
+      extends PackageGroupListEditor(PUnpopularPackages, "Unpopular Packages", "Unpopular package", parent)
+
+  private class ShunnedPackagesListEditor(parent: Composite)
+      extends PackageGroupListEditor(PShunnedPackages, "Shunned Packages", "Shunned package", parent)
+
+  private object PackageRxValidator extends IInputValidator {
+    def isValid(str: String): String = {
+      if (str == null || str.trim().isEmpty()) {
+        "Empty string is not allowed"
+      } else {
+        try {
+          str.r
+          null
+        } catch {
+          case NonFatal(e) =>
+            s"Cannot compile regular expression: ${e.getMessage}"
+        }
+      }
+    }
+  }
+}
+
+class CompletionPreferencePage extends BasicFieldEditorPreferencePage("Configure Scala Code Completion") {
+  import CompletionPreferencePage._
+
+  def createFieldEditors(): Unit = {
+    addField(new FavoritePackagesListEditor(getFieldEditorParent))
+    addField(new PreferedPackagesListEditor(getFieldEditorParent))
+    addField(new UnpopularPackagesListEditor(getFieldEditorParent))
+    addField(new ShunnedPackagesListEditor(getFieldEditorParent))
+  }
+}
+
+class ComplitionPreferenceInitializer extends AbstractPreferenceInitializer {
+  def initializeDefaultPreferences(): Unit = {
+    val store = IScalaPlugin().getPreferenceStore
+    def setDefault(key: String, regexes: Seq[Regex]): Unit = {
+      store.setDefault(key, StringListSerializer.serialize(regexes.map(_.regex)))
+    }
+
+    setDefault(PFavoritePackages, DefaultProposalRelevanceCfg.favoritePackages)
```

Comment:
`favoritePackages` is read from the pref store, but the default preferences can't be read from the pref store because the defaults is what exists first.